.github/workflows/release.yml

```diff
@@ -0,0 +1,39 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - 'v*'
+
+jobs:
+  release:
+    permissions:
+      id-token: write
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Set node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          registry-url: 'https://registry.npmjs.org'
+
+      - run: npx changelogithub
```

Comment:
We maintain our own changelog file on `CHANGELOG.md`. So this step isn't required and neither are the permissions.

Reply:
The package is mostly useful for generating a release based change log instead of file based. If we prefer files, we can do that. But for users, it will be easy to navigate across releases and see the changelogs side by side

Reply:
You can see this for reference: https://github.com/vitest-dev/vitest/releases/tag/v0.34.4